### DISCUSSION TITLE
How to write constraint to find max value among n variable

### DISCUSSION THREAD

**Discussion by dannyjpereira** [Z3Prover/z3#5409] [Q&A]
Dear sir,
I am trying to formulate SMT constraints for _max_ operation among _n_ variable. 
**max(a1,a2,.....an)**
How can we specify this function in Z3py?
Help in this regard is appreciated.

**Comment by NikolajBjorner** [comment]
1. Use nested if-then-else: max(x, y) = (if (> x y) x y)
2. Use a fresh variable that is an upper bound for a1, ..., an, assert max_x >= a1, max_x >= a2, ...
3
